contracts/stellar-axelar-gateway/src/migrate.rs

```diff
@@ -0,0 +1,89 @@
+use soroban_sdk::{Env, String, Vec};
+use stellar_axelar_std::ensure;
+use stellar_axelar_std::interfaces::CustomMigratableInterface;
+
+use crate::contract::AxelarGateway;
+use crate::error::ContractError;
+use crate::storage;
+
+pub mod legacy_storage {
+    use soroban_sdk::{contracttype, BytesN, String};
+    use stellar_axelar_std::contractstorage;
+
+    use crate::storage::MessageApprovalValue;
+
+    #[contracttype]
+    #[derive(Clone, Debug)]
+    pub struct MessageApprovalKey {
+        pub source_chain: String,
+        pub message_id: String,
+    }
+
+    #[contractstorage]
+    enum LegacyDataKey {
+        #[persistent]
+        #[value(MessageApprovalValue)]
+        MessageApproval {
+            message_approval_key: MessageApprovalKey,
+        },
+
+        #[instance]
+        #[value(u64)]
+        PreviousSignerRetention,
+
+        #[instance]
+        #[value(BytesN<32>)]
+        DomainSeparator,
+
+        #[instance]
+        #[value(u64)]
+        MinimumRotationDelay,
+
+        #[instance]
+        #[value(u64)]
+        Epoch,
```

Comment:
maybe we don't need to add the other legacy variants since they're not being used? Stellar uses the variant name so the others aren't really needed. Also helps improve code coverage